benchmarks/src/main/scala-2.12/zio/GenBenchmarks.scala

```diff
@@ -0,0 +1,92 @@
+package zio
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.scalacheck
+import zio.test.Gen
+import zio.IOBenchmarks.unsafeRun
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+class GenBenchmarks {
+
+  @Param(Array("1000"))
+  var listSize: Int = _
+
+  @Param(Array("1000"))
+  var elementSize: Int = _
+  @Benchmark
+  def zioDouble: List[Double] =
+    unsafeRun(Gen.listOfN(listSize)(Gen.uniform).sample.map(_.value).runHead.get)
+
+  @Benchmark
+  def zioIntListsOfSizeN: List[List[Int]] =
+    unsafeRun(Gen.listOfN(listSize)(Gen.listOfN(elementSize)(Gen.anyInt)).sample.map(_.value).runHead.get)
+
+  @Benchmark
+  def zioStringsOfSizeN: List[String] =
+    unsafeRun(Gen.listOfN(listSize)(Gen.stringN(elementSize)(Gen.anyChar)).sample.map(_.value).runHead.get)
+
+  @Benchmark
+  def scalaCheckDoubles: List[Double] =
+    scalacheck.Gen.listOfN(listSize, scalacheck.Gen.choose(0.0, 1.0)).sample.get
+
+  @Benchmark
+  def scalaCheckIntListsOfSizeN: List[List[Int]] =
+    scalacheck.Gen
+      .listOfN(listSize, scalacheck.Gen.listOfN(elementSize, scalacheck.Gen.choose(Int.MinValue, Int.MaxValue)))
+      .sample
+      .get
+
+  @Benchmark
+  def scalaCheckStringsOfSizeN: List[String] =
+    scalacheck.Gen
+      .listOfN(listSize, scalacheck.Gen.listOfN(elementSize, scalacheck.Gen.alphaChar).sample.mkString)
```

Comment:
Is this benchmark off because of the inner `sample` call? In the results you showed generating a string is dramatically faster than generating a list of integers of the same size which doesn't make sense.  I think it is because inside you are just generating one character instead of generating a new one each time. I think you want `.map(_.mkString)` instead of `.sample.mkString`.